File: scala-pepco-scraper/src/main/scala/com/pocketcookies/pepco/scraper/Polygon.scala
package com.pocketcookies.pepco.scraper

class Polygon(vertices: List[PointDouble]) {
  def this(s: String) = this(List.fromString(s, ' ').grouped(2).map(a => new PointDouble(a)).toList)
  def getCenter(): PointDouble = {
    val temp: PointDouble = vertices.reduceLeft((a: PointDouble, b: PointDouble) => new PointDouble(a.lat + b.lat, a.lon + b.lon))
    new PointDouble(temp.lat / vertices.size, temp.lon / vertices.size)
  }
}

File: pepco-scraper/src/main/java/com/pocketcookies/pepco/scraper/Polygon.java
package com.pocketcookies.pepco.scraper;

import com.google.common.base.Function;
import com.google.common.collect.ImmutableList;
import com.google.common.collect.Iterables;

public class Polygon {
    private final ImmutableList<PointDouble> points;

    public Polygon(String s) {
        final String[] sPoints = s.split(" ");
        assert sPoints.length % 2 == 0;
        final ImmutableList.Builder<String> builder = ImmutableList.builder();
        for (int i = 0; i < sPoints.length / 2; i++) {
            builder.add(sPoints[i] + sPoints[i + 1]);
        }
        points = ImmutableList.copyOf(Iterables.transform(builder.build(),
                new Function<String, PointDouble>() {
                    public PointDouble apply(String s) {
                        return new PointDouble(s);
                    }
                }));
    }

    public PointDouble getCenter() {
        double sumLat = 0, sumLng = 0;
        for (PointDouble pd : points) {
            sumLat += pd.lat;
            sumLng += pd.lon;
        }
        return new PointDouble(sumLat / points.size(), sumLng / points.size());
    }
}
